standards/SPIFFE_Workload_API.md

```diff
@@ -10,164 +10,354 @@ Portable and interoperable cryptographic identity for networked workloads is per
+    rpc FetchX509SVID(X509SVIDRequest) returns (stream X509SVIDResponse);
+
+    // Fetch trust bundles and CRLs. Useful for clients that only need to
+    // validate SVIDs without obtaining an SVID for themself. As this
+    // information changes, subsequent messages will be streamed from the
+    // server.
+    rpc FetchX509Bundles(X509BundlesRequest) returns (stream X509BundlesResponse);
+
+    // ... RPCS for other profiles ...
+}
+
+
+// The X509SVIDRequest message conveys parameters for requesting an X.509-SVID.
+// There are currently no such parameters.
+message X509SVIDRequest {  }
+
+// The X509SVIDResponse message carries X.509-SVIDs and related information,
+// including a global CRL and a list of bundles the workload may use for
+// federating with foreign trust domains.
 message X509SVIDResponse {
-    // A list of X509SVID messages, each of which includes a single
-    // SPIFFE Verifiable Identity Document, along with its private key
-    // and bundle.
+    // Required. A list of X509SVID messages, each of which includes a single
+    // X.509-SVID, its private key, and the bundle for the trust domain.
     repeated X509SVID svids = 1;
 
-    // ASN.1 DER encoded certificate revocation list.
+    // Optional. An ASN.1 DER encoded CRL.
     repeated bytes crl = 2;
 
-    // CA certificate bundles belonging to foreign Trust Domains that the
+    // Optional. CA certificate bundles belonging to foreign trust domains that the
     // workload should trust, keyed by the SPIFFE ID of the foreign
     // domain. Bundles are ASN.1 DER encoded.
     map<string, bytes> federated_bundles = 3;
 }
 
 // The X509SVID message carries a single SVID and all associated
-// information, including CA bundles.
+// information, including X.509 bundle for the trust domain.
 message X509SVID {
-    // The SPIFFE ID of the SVID in this entry. MUST match the SPIFFE ID
-    // encoded in the `x509_svid` certificate.
+    // Required. The SPIFFE ID of the SVID in this entry
     string spiffe_id = 1;
 
-    // ASN.1 DER encoded certificate chain. MAY include intermediates,
-    // the leaf certificate (or SVID itself) MUST come first.
+    // Required. ASN.1 DER encoded certificate chain. MAY include
+    // intermediates, the leaf certificate (or SVID itself) MUST come first.
     bytes x509_svid = 2;
 
-    // ASN.1 DER encoded PKCS#8 private key. MUST be unencrypted.
+    // Required. ASN.1 DER encoded PKCS#8 private key. MUST be unencrypted.
     bytes x509_svid_key = 3;
 
-    // CA certificates belonging to the Trust Domain
-    // ASN.1 DER encoded
+    // Required. ASN.1 DER encoded X.509 bundle for the trust domain.
     bytes bundle = 4;
+}
 
+// The X509BundlesRequest message conveys parameters for requesting X.509
+// bundles. There are currently no such parameters.
+message X509BundlesRequest {
 }
 
-// The X509BundlesResponse message carries a set of global CRLs and a
+// The X509BundlesResponse message carries a global CRL and a
```

Comment:
The `crl` field is still marked as repeated but the description here has dropped the notion of "a set of". This seems inconsistent.

Reply:
Good catch. I'll restore the original language and update the field descriptions for consistency.